### PR TITLE
IKnowMed inital Integration, EHR Administration Page and IKnowMed edit integration Drawer Form

### PR DESCRIPTION
**Pull request by ejbelville**
image

I also removed some various errors arising from the sign-up page to clean up the console.

### CODE REVIEW

File: apps/frontend/src/app/models/EHRIntegration.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+export enum EhrIntegrationType {
+    IKnowMed = 'IKnowMed',
+    Epic = 'Epic',
```
Comment (MrDiggles2) [inline apps/frontend/src/app/models/EHRIntegration.ts:3]: These should match the backend values, see `apps/backend/src/app/app-modules/ehr-integration/ehr-integration.types.ts`

```
export enum EhrIntegrationType {
  IKNOWMED = 'IKNOWMED',
  EPIC = 'EPIC'
}
```

File: apps/frontend/src/app/models/EHRIntegration.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+};
+
+export interface EhrIntegrationDB {
+    id: string;
+    platformName: string;
+    type: EhrIntegrationType;
+    config:{
+        clientId: string;
+        clientSecret: string;
+    },
```
Comment (MrDiggles2) [inline apps/frontend/src/app/models/EHRIntegration.ts:23]: Just a heads up - this `config` can be a different value based on if it's an iknowmed or epic integration

Comment (MrDiggles2) [reply apps/frontend/src/app/models/EHRIntegration.ts:23]: i.e. the epic integration looks like it'll be something like

```
config: {
    clientId: string;
    publicKey: string;
    privateKey: string;
}
```

I'll have a PR up for the backend shortly with the different types

**Review by MrDiggles2 — APPROVED** [review]
LGTM! Left some minor feedback

File: apps/frontend/src/app/pages/signup-page/ehr-integration-form/i-know-meds-form.tsx [hunk 3]
```diff
@@ -1,16 +1,52 @@
...
+  const currentOrganization = organizations.find((org => user?.org_id === org.id ));
+
+
+  const [form, setForm] = React.useState<EhrIntegration>({
+    platformName: '',
+    clientId: '',
+    clientSecret: '',
+    id: undefined,
+    type: EhrIntegrationType.IKnowMed,
+    organizationId: currentOrganization?.id,
```
Comment (MrDiggles2) [inline apps/frontend/src/app/pages/signup-page/ehr-integration-form/i-know-meds-form.tsx:23]: I'm getting a couple of instances where organization loads after this state is set which causes the API request to fail due to missing org ID.

Can we update this to be set via an effect? Something like

```
useEffect(() => {
    setForm({...form, organizationId: currentOrganization.id})
}, [currentOrganization])
```

**Review by MrDiggles2 — CHANGES_REQUESTED** [review]

File: apps/frontend/src/app/models/EHRIntegration.ts (hunk 2 above)
Comment (ejbelville) [reply apps/frontend/src/app/models/EHRIntegration.ts:29]: Great, I'm working on adding the Epic Form now and am handling the different types as I come across them. So this'll be updated in my next PR.

File: apps/frontend/src/app/pages/signup-page/ehr-integration-form/i-know-meds-form.tsx (hunk 3 above)
Comment (ejbelville) [reply apps/frontend/src/app/pages/signup-page/ehr-integration-form/i-know-meds-form.tsx:23]: Added

File: apps/frontend/src/app/models/EHRIntegration.ts (hunk 1 above)
Comment (ejbelville) [reply apps/frontend/src/app/models/EHRIntegration.ts]: Done!

**Review by MrDiggles2 — APPROVED** [review]
LGTM! 🚢

**Pull request merged**